### ISSUE TITLE
The metadata type of `ExpressionContextProvider` should be `ExpressionContext` instead of `Optional[ExpressionContext]`

### ISSUE THREAD

**Issue by MapleCCC** [Instagram/LibCST#758]
Looking at the [implementation code](https://github.com/Instagram/LibCST/blob/main/libcst/metadata/expression_context_provider.py) of `ExpressionContextProvider`, we can conclude that every invocation of `set_metadata()` inserts an `ExpressionContext` object into the metadata table. It never once inserts a `None` object.

Therefore we are safe to narrow down the metadata type for it. I can imagine a more precise type being beneficial for downstream code, such as reducing unnecessary `is not None` checks everywhere.

**Comment by zsol** [comment]
Yep, seems reasonable. There's no use for a `None` expression context; if the metadata can be determined for a node, it must be a valid `ExpressionContext` value.

**Comment by MapleCCC** [comment]
Nice! A PR #760 is opened to address the proposed change.
